batchservice/src/main/java/nu/yona/server/batch/jobs/ActivityAggregationBatchJob.java

```diff
@@ -0,0 +1,176 @@
+	private JpaPagingItemReader<WeekActivity> weekActivityReader;
+
+	@Bean("activityAggregationJob")
+	public Job activityAggregationBatchJob()
+	{
+		return jobBuilderFactory.get("activityAggregationBatchJob").incrementer(new RunIdIncrementer())
+				.flow(aggregateDayActivities()).next(aggregateWeekActivities()).end().build();
+	}
+
+	private Step aggregateDayActivities()
+	{
+		return stepBuilderFactory.get("aggregateDayActivities").<DayActivity, DayActivity> chunk(DAY_ACTIVITY_CHUNK_SIZE)
+				.reader(dayActivityReader).processor(dayActivityProcessor()).writer(dayActivityWriter()).build();
+	}
+
+	private Step aggregateWeekActivities()
+	{
+		return stepBuilderFactory.get("aggregateWeekActivities").<WeekActivity, WeekActivity> chunk(WEEK_ACTIVITY_CHUNK_SIZE)
+				.reader(weekActivityReader).processor(weekActivityProcessor()).writer(weekActivityWriter()).build();
+	}
+
+	@Bean(name = "activityAggregationJobDayActivityReader", destroyMethod = "")
+	@StepScope
+	public JpaPagingItemReader<DayActivity> dayActivityReader()
+	{
+		try
+		{
+			String jpqlQuery = "SELECT d FROM DayActivity d WHERE d.startDate <= :yesterday AND d.aggregatesComputed = false";
+
+			JpaPagingItemReader<DayActivity> reader = new JpaPagingItemReader<>();
+			reader.setQueryString(jpqlQuery);
+			reader.setParameterValues(
+					Collections.singletonMap("yesterday", ZonedDateTime.now(DEFAULT_TIME_ZONE).minusDays(1).toLocalDate()));
+			reader.setEntityManagerFactory(entityManager.getEntityManagerFactory());
+			reader.setPageSize(DAY_ACTIVITY_CHUNK_SIZE);
+			reader.afterPropertiesSet();
+			reader.setSaveState(true);
+
+			return reader;
+		}
+		catch (Exception e)
+		{
+			throw YonaException.unexpected(e);
+		}
+	}
+
+	private ItemProcessor<DayActivity, DayActivity> dayActivityProcessor()
+	{
+		return new ItemProcessor<DayActivity, DayActivity>() {
+			@Override
+			public DayActivity process(DayActivity dayActivity) throws Exception
+			{
+				logger.debug("Processing day activity with id {}, start date {}", dayActivity.getId(),
+						dayActivity.getStartDate());
+
+				dayActivity.computeAggregates();
+				return dayActivity;
+			}
+		};
+	}
+
+	private JpaItemWriter<DayActivity> dayActivityWriter()
+	{
+		JpaItemWriter<DayActivity> writer = new JpaItemWriter<>();
+		writer.setEntityManagerFactory(entityManager.getEntityManagerFactory());
+
+		return writer;
+	}
+
+	@Bean(name = "activityAggregationJobWeekActivityReader", destroyMethod = "")
+	@StepScope
+	public JpaPagingItemReader<WeekActivity> weekActivityReader()
+	{
+		try
+		{
+			String jpqlQuery = "SELECT w FROM WeekActivity w WHERE w.startDate <= :lastWeek AND w.aggregatesComputed = false";
+
+			JpaPagingItemReader<WeekActivity> reader = new JpaPagingItemReader<>();
+			reader.setQueryString(jpqlQuery);
+			reader.setParameterValues(Collections.singletonMap("lastWeek",
+					TimeUtil.getStartOfWeek(ZonedDateTime.now(DEFAULT_TIME_ZONE).toLocalDate()).minusWeeks(1)));
```

Comment:
Thanks for pointing this out. I also forgot to use `TimeUtil.getStartOfDay` in the other case, so fixed that too.